### ISSUE TITLE
EXIF datetime format is unsupported

### ISSUE THREAD

**Issue by piepmatz** [dateutil/dateutil#271]
Photos with EXIF data contain date and time when the image was taken.
The format is YYYY:MM:DD HH:MM:SS. This results in an incorrectly parsed datetime.

Example:
```python
>>> from dateutil.parser import parse
>>> parse("2015:04:15 17:15:34")
datetime.datetime(2016, 5, 25, 17, 15, 34)

>>> from datetime import date
>>> date.today()
datetime.date(2016, 5, 25)
```

The example shows that the time is parsed correctly, but the date is replaced by the current date.

Tested with dateutil version 2.5.3.

The EXIF specification is over here: <http://www.cipa.jp/std/documents/e/DC-008-2012_E.pdf>
The format is specified on pages (PDF count) 44, 53, 54.

**Comment by labrys** [comment]
@piepmatz if you know the format of the data and since there is no timezone information, why not use [`datetime.strptime()`](https://docs.python.org/2/library/datetime.html#strftime-and-strptime-behavior)?

**Comment by pganssle** [comment]
I agree that if you happen to know the format, regardless of whether or not `dateutil.parser.parse` works, the right thing to do is to use `datetime.strptime`.

That said, I do think that parsing these types of dates is within the scope of the parser, which is intended to parse anything that reasonably looks like a datetime, and if something's a widely used standardized format, that probably reasonably looks like a datetime.

I think this should be addressed as part of the upcoming "parser overhaul", which I'm thinking will be pushed to 2.7.0 rather than 2.6.0, because it's probably better to get the time zone improvements out sooner rather than later.
